spec/apis/path-to-active-when.js

```diff
@@ -0,0 +1,145 @@
+import { pathToActiveWhen } from "single-spa";
+import { toDynamicPathValidatorRegex } from "../../src/applications/apps";
+
+describe(`pathToActiveWhen`, () => {
+  it(`allows you to create your own activeWhen functions using the single-spa public api`, () => {
+    const activeWhen = pathToActiveWhen("/users/:userId/settings");
+
+    expect(activeWhen(new URL("http://localhost/users/1/settings"))).toBe(true);
+    expect(activeWhen(new URL("http://localhost/users/1"))).toBe(false);
+    expect(activeWhen(new URL("http://localhost/users"))).toBe(false);
+    expect(activeWhen(new URL("http://localhost/cart"))).toBe(false);
+  });
+
+  describe("toDynamicPathValidatorRegex", () => {
```

Comment:
This test suite was moved from the old test file to this new one. It tests internal functions, not a public API. But now the outer wrapper function for toDynamicPathValidatorRegex is exposed as a public API, so I decided to merge the two tests together.